tubular/imputers.py

```diff
@@ -461,40 +463,41 @@ def fit(self, X: pd.DataFrame, y: pd.Series) -> pd.DataFrame:
 
         super().fit(X, y)
 
-        n_nulls = y.isna().sum()
+        n_nulls = y.is_null().sum()
 
         if n_nulls > 0:
             msg = f"{self.classname()}: y has {n_nulls} null values"
             raise ValueError(msg)
 
         self.impute_values_ = {}
 
-        X_y = self._combine_X_y(X, y)
+        X_y = nw.from_native(self._combine_X_y(X, y))
         response_column = "_temporary_response"
 
         for c in self.columns:
-            c_nulls = X[c].isna()
+            c_nulls = X[c].is_null()
 
             if c_nulls.sum() == 0:
                 msg = f"{self.classname()}: Column {c} has no missing values, cannot use this transformer."
                 raise ValueError(msg)
 
-            mean_response_by_levels = pd.DataFrame(
-                X_y.loc[~c_nulls].groupby(c)[response_column].mean(),
-            ).reset_index()
+            mean_response_by_levels = (
+                X_y.filter(~c_nulls).group_by(c).agg(nw.col(response_column).mean())
+            )
 
-            mean_response_nulls = X_y.loc[c_nulls, response_column].mean()
+            mean_response_nulls = X_y.filter(c_nulls)[response_column].mean()
 
-            mean_response_by_levels["abs_diff_response"] = np.abs(
-                mean_response_by_levels[response_column] - mean_response_nulls,
+            mean_response_by_levels = mean_response_by_levels.with_columns(
+                abs_diff_response=np.abs(
```

Comment:
S: would this work like:
```
mean_response_by_levels=mean_response_by_levels.with_columns(
(nw.col(response_column)-mean_response_nulls).abs().alias('abs_diff_response')
)
```

Reply:
[here](https://github.com/lvgig/tubular/pull/342/commits/01de73607f1ef09031fb121b17b6b4bdc36c9212)